alicloud/resource_alicloud_cms_sitemonitor.go

```diff
@@ -0,0 +1,190 @@
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"update_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCmsSiteMonitorCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	cmsService := CmsService{client}
+
+	taskName := d.Get("task_name").(string)
+	request := cms.CreateCreateSiteMonitorRequest()
+	request.Address = d.Get("address").(string)
+	request.TaskName = taskName
+	request.TaskType = d.Get("task_type").(string)
+	request.Interval = strconv.Itoa(d.Get("interval").(int))
+	request.OptionsJson = d.Get("options_json").(string)
+	alertIds := d.Get("alert_ids").([]interface{})
+	alertId := getAlertId(alertIds)
+	if alertId != "" {
+		request.AlertIds = alertId
+	}
+
+	_, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
+		return cmsClient.CreateSiteMonitor(request)
+	})
+	if err != nil {
+		return WrapError(err)
+	}
+
+	siteMonitor, err := cmsService.DescribeSiteMonitor("", taskName)
+	if err != nil {
+		return WrapError(err)
+	}
+
+	d.SetId(siteMonitor.TaskId)
+
+	return resourceAlicloudCmsSiteMonitorRead(d, meta)
+}
+
+func resourceAlicloudCmsSiteMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	cmsService := CmsService{client}
+
+	siteMonitor, err := cmsService.DescribeSiteMonitor(d.Id(), "")
+	if err != nil {
+		return WrapError(err)
+	}
+
+	d.Set("address", siteMonitor.Address)
+	d.Set("task_name", siteMonitor.TaskName)
+	d.Set("task_type", siteMonitor.TaskType)
+	d.Set("task_state", siteMonitor.TaskState)
+	d.Set("interval", siteMonitor.Interval)
+	d.Set("options_json", siteMonitor.OptionsJson)
+	d.Set("create_time", siteMonitor.CreateTime)
+	d.Set("update_time", siteMonitor.UpdateTime)
+
+	return nil
+}
+
+func resourceAlicloudCmsSiteMonitorUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := cms.CreateModifySiteMonitorRequest()
+	request.TaskId = d.Id()
+	request.Address = d.Get("address").(string)
+	request.Interval = strconv.Itoa(d.Get("interval").(int))
+	request.OptionsJson = d.Get("options_json").(string)
+	request.TaskName = d.Get("task_name").(string)
+	alertIds := d.Get("alert_ids").([]interface{})
+	alertId := getAlertId(alertIds)
+	if alertId != "" {
+		request.AlertIds = alertId
+	}
+
+	_, err := client.WithCmsClient(func(cmsClient *cms.Client) (interface{}, error) {
+		return cmsClient.ModifySiteMonitor(request)
+	})
+	if err != nil {
+		return WrapError(err)
+	}
+
+	return resourceAlicloudCmsSiteMonitorRead(d, meta)
+}
+
+func resourceAlicloudCmsSiteMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	cmsService := CmsService{client}
+	request := cms.CreateDeleteSiteMonitorsRequest()
+
+	request.TaskIds = d.Id()
+	request.IsDeleteAlarms = "false"
```

Comment:
Why the line 160 sets hard code 'false"? Does it can expose to a parameter?

Reply:
this is only if you want to delete alarms that have been associated to this sitemonitor. Since alarms are created separately and have their own resource provider I think their lifecycle should not be managed by this flag.